src/Services/GarbageCollectionService.php

```diff
@@ -2,22 +2,50 @@
 
 namespace SilverStripe\SessionManager\Services;
 
+use SilverStripe\Core\Config\Configurable;
 use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\FieldType\DBDatetime;
 use SilverStripe\Security\RememberLoginHash;
 use SilverStripe\SessionManager\Models\LoginSession;
 
 class GarbageCollectionService
 {
+    use Configurable;
     use Injectable;
 
+    /**
+     * Limit the number of records collected per run. Defaults to 0 (no limit).
+     */
+    private static int $batch_remove_limit = 0;
+
     /**
      * Delete expired LoginSession and RememberLoginHash records
      */
-    public function collect(): void
+    public function collect(): array
     {
-        $this->collectExpiredSessions();
-        $this->collectImplicitlyExpiredSessions();
-        $this->collectExpiredLoginHashes();
+        $expiredSessions = $this->collectExpiredSessions();
+        $implicitlyExpiredSessions = $this->collectImplicitlyExpiredSessions();
+        $loginHashes = $this->collectExpiredLoginHashes();
+        return [
+            'expiredSessions' => $expiredSessions,
+            'implicitlyExpiredSessions' => $implicitlyExpiredSessions,
+            'loginHashes' => $loginHashes,
+        ];
+    }
+
+    private function batchRemoveAll($datalist): int
+    {
+        $i = 0;
+        $limit = self::config()->get('batch_remove_limit');
+        $limitedList = $limit > 0 ? $datalist->limit($limit) : $datalist;
```

Comment:
```suggestion
        $limit = self::config()->get('batch_remove_limit');
        $limitedList = $datalist->limit($limit);
```
With the default limit set to null, we can just trust the limit value outright.